### PR TITLE
Base sample

### PR DESCRIPTION
**Pull request by Quentin01**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,10 @@ macro(lug_set_option var default type docstring)
     set(${var} ${${var}} CACHE ${type} ${docstring} FORCE)
 endmacro()
 
+if (${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
+    message( FATAL_ERROR "In-source builds not allowed. Please make a new directory (called a build directory) and run CMake from there. You may need to remove CMakeCache.txt." )
```
Comment (saveman71) [inline CMakeLists.txt]: Remove spaces around parenthesis

File: cmake/modules/FindLUG.cmake [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+#
+# If Lugdunum is not installed in a standard path, you can use the LUG_ROOT CMake (or environment) variable
+# to tell CMake where Lugdunum is.
+#
+# Output
+# ------
+#
+# This script defines the following variables:
+# - For each specified module XXX (system, window, graphics, core, etc):
+#   - LUG_XXX_LIBRARY_DEBUG:   the name of the debug library of the xxx module (set to LUG_XXX_LIBRARY_RELEASE is no debug version is found)
```
Comment (saveman71) [inline cmake/modules/FindLUG.cmake:30]: Be consistent with XXX or xxx

File: cmake/modules/FindLUG.cmake [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                     LUG_${FIND_LUG_COMPONENT_UPPER}_LIBRARY_DYNAMIC_RELEASE
+                     LUG_${FIND_LUG_COMPONENT_UPPER}_LIBRARY_DYNAMIC_DEBUG
+    )
+
+    # add to the global list of libraries
+    set(LUG_LIBRARIES ${LUG_LIBRARIES} "${LUG_${FIND_LUG_COMPONENT_UPPER}_LIBRARY}")
+
+endforeach()
+
+if(NOT LUG_FOUND)
```
Comment (saveman71) [inline cmake/modules/FindLUG.cmake]: Can't you just merge this `if` branch with the `else()` clause below?

File: samples/base/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+endif()
+
+# project name
+project(base)
+
+# define the executable
+set(EXECUTABLE_NAME "base")
+add_executable(${EXECUTABLE_NAME} main.cpp)
+
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../../cmake//modules" ${CMAKE_MODULE_PATH})
```
Comment (saveman71) [inline samples/base/CMakeLists.txt]: Double slash?

Comment (saveman71) [reply samples/base/CMakeLists.txt]: Also `list(APPEND ${CMAKE_MODULE_PATH} value)`

File: samples/base/README.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# Base sample
+This is just a base sample to see how to compile your project with Lugdunum.
```
Comment (saveman71) [inline samples/base/README.md]: s/too see/that shows/

**Review by saveman71 — CHANGES_REQUESTED** [review]

File: cmake/modules/FindLUG.cmake (hunk 2 above)
Comment (Quentin01) [reply cmake/modules/FindLUG.cmake:30]: This is the attended case, the lib which is called "system" will be named "SYSTEM" in the variable names.

Comment (saveman71) [reply cmake/modules/FindLUG.cmake:30]: Oh okay. *intended.

**Review by saveman71 — APPROVED** [review]

**Pull request merged**